src/foam/mlang/predicate/CapabilityAuthServicePredicate.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.predicate',
+  name: 'CapabilityAuthServicePredicate',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  documentation: `
+    Create modeled predicate.
+  `,
+
+  javaImports: [
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'foam.nanos.logger.Logger',
+  ],
+
+  properties: [
+    {
+      name: 'capabilityDAO',
+      class: 'FObjectProperty',
+      of: 'foam.dao.DAO'
+    },
+    {
+      name: 'permission',
+      class: 'String'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      args: [
+        {
+          name: 'obj',
+          type: 'Object'
+        }
+      ],
+      javaCode: `
+          foam.core.X x = getX();
+          UserCapabilityJunction ucj = (UserCapabilityJunction) obj;
+          if ( ucj.getStatus() == CapabilityJunctionStatus.GRANTED ) {
+            Capability c = (Capability) getCapabilityDAO().find(ucj.getTargetId());
+            if ( c != null &&
```

Comment:
This indentation is weird and hard to read, just put the whole expression on one line as it fits easily.